protocols/imap/src/test/java/org/apache/james/imap/processor/AbstractSelectionProcessorTest.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.processor;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.apache.james.imap.api.message.IdRange;
+import org.apache.james.imap.api.message.UidRange;
+import org.apache.james.imap.api.message.response.StatusResponseFactory;
+import org.apache.james.imap.api.process.ImapProcessor;
+import org.apache.james.imap.api.process.SelectedMailbox;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.metrics.api.MetricFactory;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class AbstractSelectionProcessorTest {
+    private SelectProcessor testee;
+
+    @Before
+    public void setUp() {
+        ImapProcessor nextProcessor = null;
+        MailboxManager mailboxManager = null;
+        StatusResponseFactory statusResponseFactory = null;
+        MetricFactory metricFactory = null;
+        testee = new SelectProcessor(nextProcessor, mailboxManager, statusResponseFactory, metricFactory);
+    }
+
+    @Ignore("JAMES-2278 Processing fails with an index out of bound")
+    @Test
+    public void recomputeUidSetShouldNotFailWhenDataIsProvided() {
```

Comment:
I don't think that "data is provided" gives a precise enough description of this test case.
And to be honest, I don't manage to understand what the test case is.

Reply:
Well, as soon as the code iterates on the array, it throwed.

So providing a non empty range trigger the bug.

Reply:
Ok, what about recomputeUidSetShouldNotFailWhenNonEmptyRange ?